src/main/java/org/apache/sling/installer/provider/jcr/impl/JcrUtil.java

```diff
@@ -73,4 +75,21 @@ public static Node createPath(final Session session,
         }
         return parentNode.getNode(relativePath);
     }
+
+    /**
+     * Get the PID for a configuration event by using the R7 format before saving it.
+     *
+     * @param factoryPid factory PID of the configuration
+     * @param pid PID of the configuration
+     * @return The PID in R7 format
+     */
+    public static String getPid(final String factoryPid, final String pid) {
+        // if factory pid is separated from pid by a period (.), we need replace it with a ~ so that this can be installed
+        // and grouped as a factory configuration
+        if (pid.startsWith(factoryPid + '.')) {
+            String id = pid.substring(factoryPid.length() + 1);
+            return factoryPid + "~" + id;
+        }
+        return pid;
```

Comment:
I would flip the logic. I.e. let a pid in the correct format through. If the format is incorrect, normalize the pid to a name by removing a prefix of `factoryPid` if present. Then concat the factoryPid, ~ and normalized pid.

Reply:
Addressed in https://github.com/apache/sling-org-apache-sling-installer-provider-jcr/pull/9/commits/6b9cdd225224636759cd64a96328c9b573316aff

Reply:
I was thinking to keep this slightly more generic (note: untested code below!):

```suggestion
        final String name;
        if (pid.matches("^" + factoryPid + "[^a-zA-Z0-9\\s]")) {
            name = pid.substring(factoryPid.length() + 1);
        } else {
            name = pid;
        }
        return factoryPid + "~" + name;
```

WDYT?

Reply:
Addressed in https://github.com/apache/sling-org-apache-sling-installer-provider-jcr/pull/9/commits/49da45e49774d6f9b8597bbc6376c527ef5970f3